tests/test_builders/test_build_linkcheck.py

```diff
@@ -843,7 +849,7 @@ def test_TooManyRedirects_on_HEAD(app, monkeypatch):
     }
 
 
-def make_retry_after_handler(responses):
+def make_retry_after_handler(responses: list[tuple[int, str | None]]) -> type[BaseHTTPRequestHandler]:
```

Comment:
Probably technically an `Iterable[tuple[int, str | None]]` instead of a list here (a series of responses that can be emitted in sequence, although not all of them may be consumed).